index.ts

```diff
@@ -374,7 +374,8 @@ export function poseidonHashMany(values: bigint[], fn = poseidonSmall): bigint {
     for (let j = 0; j < rate; j++) {
       const item = padded[i + j];
       if (typeof item === 'undefined') throw new Error('invalid index');
-      state[j] += item;
+      if (typeof state[j] === 'undefined') throw new Error('state[j] is undefined');
+      state[j] = state[j]! + item;
```

Comment:
Although the TS warning looks very much like a bug in TS `!` type assertion is still required even after the explicit `typeof state[j] === 'undefined'` check